src/bitcoin-cli.cpp

```diff
@@ -148,7 +148,11 @@ static void http_request_done(struct evhttp_request *req, void *ctx)
 UniValue CallRPC(const string& strMethod, const UniValue& params)
 {
     std::string host = GetArg("-rpcconnect", DEFAULT_RPCCONNECT);
-    int port = GetArg("-rpcport", BaseParams().RPCPort());
+
+    int portTmp = GetArg("-rpcport", BaseParams().RPCPort());
+    if(portTmp <= 0 || portTmp >= 0x10000)
```

Comment:
That would require:
1. include netbase.h
2. move IsValidPort definition to netbase.h
Not sure it worth it tbh.